src/services/completions.ts

```diff
@@ -852,32 +852,34 @@ namespace ts.Completions {
         let isInSnippetScope = false;
         if (insideComment) {
             if (hasDocComment(sourceFile, position)) {
-                if (sourceFile.text.charCodeAt(position - 1) === CharacterCodes.at) {
-                    // The current position is next to the '@' sign, when no tag name being provided yet.
-                    // Provide a full list of tag names
-                    return { kind: CompletionDataKind.JsDocTagName };
-                }
-                else {
-                    // When completion is requested without "@", we will have check to make sure that
-                    // there are no comments prefix the request position. We will only allow "*" and space.
-                    // e.g
-                    //   /** |c| /*
-                    //
-                    //   /**
-                    //     |c|
-                    //    */
-                    //
-                    //   /**
-                    //    * |c|
-                    //    */
-                    //
-                    //   /**
-                    //    *         |c|
-                    //    */
-                    const lineStart = getLineStartPositionForPosition(position, sourceFile);
-                    if (!(sourceFile.text.substring(lineStart, position).match(/[^\*|\s|(/\*\*)]/))) {
-                        return { kind: CompletionDataKind.JsDocTag };
-                    }
+                // When completion is requested without "@", we will have check to make sure that
+                // there are no comments prefix the request position. We will only allow "*" and space.
+                // e.g
+                //   /** |c| */
+                //
+                //   /**
+                //     |c|
+                //    */
+                //
+                //   /**
+                //    * |c|
+                //    */
+                //
+                //   /**
+                //    *         |c|
+                //    */
+                const lineStart = getLineStartPositionForPosition(position, sourceFile);
+                const jsdocFragment = sourceFile.text.substring(lineStart, position);
+                const reJSDocFragment = /^(?:\s*\/\*\*\s+|\s+\*?\s+)(@(?:\w+)?)?/g;
+                const match = reJSDocFragment.exec(jsdocFragment);
+                if (match && reJSDocFragment.lastIndex === jsdocFragment.length) {
+                    return {
+                        kind: match[1]
+                        // The current position is next to the '@' sign, when no tag name being provided yet.
+                        // Provide a full list of tag names
+                        ? CompletionDataKind.JsDocTagName:

```

Comment:
it makes sense to break this line so you can document JsDocTagName. Can you move the trailing `:` to the next line though?

Reply:
@sandersn hello :-)

Would this be okay?
```ts
    return {
        kind: match[1] ?
            // The current position is next to the '@' sign, when no tag name being provided yet.
            // Provide a full list of tag names
            CompletionDataKind.JsDocTagName
            : CompletionDataKind.JsDocTag
    };
```